manim/scene/stream_file_writer.py

```diff
@@ -0,0 +1,103 @@
+from abc import ABCMeta
+
+import os
+import subprocess
+
+from .. import config, logger
+from ..constants import FFMPEG_BIN
+from .scene import Scene
+from .scene_file_writer import SceneFileWriter
+from ..utils.file_ops import guarantee_existence
+
+
+class StreamFileWriter(SceneFileWriter):
+    def __init__(self, renderer):
+        super().__init__(renderer, "")
+        vars(self).update(config.streaming_config)
+        path = os.path.join(config.get_dir("streaming_dir"), "clips")
+        self.FOLDER_PATH = os.path.relpath(guarantee_existence(path))
+        # To prevent extensive overwriting
+        self.partial_movie_directory = self.FOLDER_PATH
+
+    def init_output_directories(self, scene_name):
+        """The original :class:`SceneFileWriter` uses this method while initializing.
+        The method is overwritten carefully so as to correspond arguments in the
+        original and still avoid the folder creation effects of this method.
+        """
+        pass
+
+    @property
+    def file_path(self):
+        return self.partial_movie_files[-1]
+
+    def end_animation(self, allow_write=False):
+        """The point in the animation where the file exists."""
+        super().end_animation(allow_write=allow_write)
+        self.stream()
+
+    def combine_movie_files(self):
+        """Also to reduce overriding code."""
+        pass
+
+    def stream(self):
+        logger.info(
+            "Houston, we are ready to launch. Sending over to %(url)s",
+            {"url": {self.streaming_url}},
+        )
+        command = [
+            FFMPEG_BIN,
+            "-re",
+            "-i",
+            self.file_path,
+            "-vcodec",
+            "copy",
+            "-an",
+            "-loglevel",
+            "quiet",
+        ]
+
+        if self.streaming_protocol == "rtp":
+            command += ["-sdp_file", self.sdp_path]
+        command += [
+            "-f",
+            self.streaming_protocol,  # Take a look here for other streaming protocols
+            self.streaming_url,
+        ]
+        os.system(" ".join(command))
+
+    def open_movie_pipe(self):
```

Comment:
The amount of code duplication between here and `SceneFileWriter.open_movie_pipe` is unfortunate, and could be reduced substantially by refactoring a little. I won't block the PR because of it, however.

Please add a one-line docstring summary, suggestion:

```suggestion
    def open_movie_pipe(self):
        """Used internally by Manim to initialise
        FFMPEG and begin writing to FFMPEG's input
        buffer.
        """
```

Reply:
Oh yeah, that bit me a little that that code looks eerily similar to the one from the inherited class. That one I noticed as I was coding.